### PR TITLE
Add test to verify virtual device usage metrics

### PR DESCRIPTION
**Pull request by steventk-g**

### CODE REVIEW

**Review by yeounoh — CHANGES_REQUESTED** [review]
LGTM, requested to add another quick test case.

File: test/test_xla_sharding.py [hunk 1]
```diff
@@ -137,6 +138,15 @@ def test_mark_sharding(self):
...
                          device=xm.xla_device())))
 
+  def test_metrics_recorded(self):
+    partition_spec = (0, 1)
+    xt1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
+                       dtype=torch.float,
+                       device=xm.xla_device())
+    xs.mark_sharding(xt1, self._get_mesh((1, self.n_devices)), partition_spec)
+    self.assertIn("VirtualDeviceUsage", met.counter_names())
+    self.assertNotEqual(met.counter_value("VirtualDeviceUsage"), 0)
```
[the remark this answers is not in the document]
Comment (steventk-g) [reply test/test_xla_sharding.py:150]: Added a test for sharding on model weights. It looks like virtual device will still be used to delay the transfer of the model weights in `nn.Linear(128, 64).to(xm.xla_device())` until the sharding is applied.

[the remark this answers is not in the document]
Comment (yeounoh) [reply test/test_xla_sharding.py:150]: Nice!

**Review by yeounoh — APPROVED** [review]
LGTM

**Review by JackCaoG — APPROVED** [review]

**Pull request merged**